types/src/transaction/signature_verified_transaction.rs

```diff
@@ -26,6 +34,13 @@ impl SignatureVerifiedTransaction {
         }
     }
 
+    pub fn borrow_into_inner(&self) -> &Transaction {
```

Comment:
It stinks that a valid txn and an invalid one can equal..

how about make it `expect_valid_inner_ref()` and panic on the Invalid flavor (are we sure txns are all valid where you compare them?)

Reply:
Yes this makes sense. I think i'll change the `PartialEq` implementation actually to check that they're either:
1. If Both valid -> return inner equality check
2. If Both invalid -> return inner equality check
3. Else -> panic